### ISSUE TITLE
Nested list operations wrongfully raises require_trailing_commas

### ISSUE THREAD

**Issue by TheCarpetMerchant** [dart-lang/linter#3948]
See the [gist](https://gist.github.com/TheCarpetMerchant/4cfa30dc19d22c7a225666f2214b11e7).
require_trailing_commas is raised for each end of line after `total`.

**Comment by lauweijie** [comment]
Can you explain why the trailing comma is unexpected after each `total`?

**Comment by TheCarpetMerchant** [comment]
@lauweijie Because `total,.total` is not valid dart syntax.

**Comment by lrhn** [comment]
I think the `,` should go after the `.total`, which is where the argument expression ends.

The lint description states
> **DO** use trailing commas for all function calls and declarations unless the function call or definition, from the start of the function name up to the closing parenthesis, fits in a single line.

All but one of those calls satisfy that.
The exception does not apply.
> **EXCEPTION**: If the final parameter/argument is positional (vs named) and is either a function literal implemented using curly braces, a literal map, a literal set or a literal array. This exception only applies if the final parameter does not fit entirely on one line.

The expression is a `something.map(...).total`, which is not one of the exceptions.

**Comment by lauweijie** [comment]
Agree with @lrhn, the `,` goes after `.total` and not `total),.total`.

Per your example, it should be something like this after dartfmt:
```
Duration getTotal() {
  return readingTimes.entries.map(
    (a) => a.value.entries.map(
      (b) => b.value.entries.map(
        (c) => c.value.entries.map(
          (d) => d.value.entries.map(
            (e) => e.value.entries.map(
              (f) => f.value.entries.map((g) => g.value).total,
            ).total,
          ).total,
        ).total,
      ).total,
    ).total,
  ).total;
}
```

**Comment by TheCarpetMerchant** [comment]
I have no idea why, but the Android Studio auto-fix hadn't put them there. Just re-tested it and it's indeed fine, my bad.

However, it does make it look absolutely horrendous after dartfmt :
```
Duration getTotal() {
    return readingTimes.entries
        .map(
          (a) => a.value.entries
              .map(
                (b) => b.value.entries
                    .map(
                      (c) => c.value.entries
                          .map(
                            (d) => d.value.entries
                                .map(
                                  (e) => e.value.entries
                                      .map(
                                        (f) => f.value.entries
                                            .map((g) => g.value)
                                            .total,
                                      )
                                      .total,
                                )
                                .total,
                          )
                          .total,
                    )
                    .total,
              )
              .total,
        )
        .total;
  }
```

So I'm just gonna put an //ignore_for_file. Thanks !
